docs/source/architecture.rst

```diff
@@ -1,19 +1,740 @@
+
+Pseudo-random workload distribution can cause distribution anomalies within a workload control cycle but it provides a relatively smooth distribution on longer timescales, without the need for full visibility and operations aggregation.
+
+Such a workload distribution and the allocation functions for :ref:`dynamic roles <dynamic_roles>` are the core instruments that enable the :ref:`omni-scaling <omniscaling>` feature of the Insolar Platform. This feature provides a balance in accordance with client's needs.
+
+Processing costs can be traded off against:
+
+* **Uninsured risks**. Suitable for situations where a cheaper transaction is executed but fewer validators verify said transaction, meaning greater risk of loss.
+* **Processing speed**. It can be increased to the detriment of operational risk:
+
+  * frequent transactions could be processed without awaiting validation, or
+  * validations may be batched together and processed following some delay, leading to the possibility of resource-consuming rollbacks.
+
+.. _execution_validation:
+
+Execution & Validation
+~~~~~~~~~~~~~~~~~~~~~~
+
+The Insolar Platform works on the principle of actions executed by one node, validated by many.
+
+The number of elected validators can be determined in accordance with the :ref:`business process <domains>` at hand and, since validators in shared enterprise networks will have liability and legal guarantees, this works as transaction insurance.
+
+As described in the :ref:`network consistency section <network_consistency>`, validator elections are *not* based on voting; instead, they are part of the :ref:`omni-scaling <omniscaling>` feature. Insolar uses the active node list and :ref:`entropy <pulsars>` generated by consensus of the :ref:`globula network protocol <network_consensus>`, and then applies deterministic allocation functions for :ref:`node roles <dynamic_roles>`. This avoids wasting efforts on numerous per-transaction and network-wide consensuses.
+
+Since Insolar sets apart functionality using :ref:`node roles <multi_role_nodes>`, it has two sets of execution & validation procedures: **virtual** and **material**.
+
+.. _virtual_execution_validation:
+
+Virtual Execution & Validation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Nodes with :ref:`virtual static roles <virtual>` carry out **virtual** execution & validation:
+
+#. The network selects (determines based on :term:`entropy <pulse>`) a specific virtual node to become a :ref:`virtual executor <dynamic_roles>`. Upon receiving the request, the executor:
+
+   #. Registers the request within the current :term:`pulse`.
+
+      In case the request refers to a 'busy' :term:`object <object>`, execution may be delayed and handed over to another executor due to state locks or multiple updates. Moreover, multiple requests can be executed within the same pulse when opportunistic execution/validation is allowed by the caller or by the called object.
```

Comment:
In case the request ARRIVEs to a 'busy' virtual executor, it can delegate execution of some object to other virtual nodes (not necessary to virtual executors)